generate/generate.go

```diff
@@ -0,0 +1,1015 @@
+// Package generate implements functions generating container config files.
+package generate
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/syndtr/gocapability/capability"
+)
+
+var (
+	// Namespaces include the names of supported namespaces.
+	Namespaces = []string{"network", "pid", "mount", "ipc", "uts", "user", "cgroup"}
+)
+
+// Generator represents a generator for a container spec.
+type Generator struct {
+	spec *rspec.Spec
+}
+
+// New creates a spec Generator with the default spec.
+func New() Generator {
+	spec := rspec.Spec{
+		Version: rspec.Version,
+		Platform: rspec.Platform{
+			OS:   runtime.GOOS,
+			Arch: runtime.GOARCH,
+		},
+		Root: rspec.Root{
+			Path:     "",
+			Readonly: false,
+		},
+		Process: rspec.Process{
+			Terminal: false,
+			User:     rspec.User{},
+			Args: []string{
+				"sh",
+			},
+			Env: []string{
+				"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+				"TERM=xterm",
+			},
+			Cwd: "/",
+			Capabilities: []string{
+				"CAP_CHOWN",
+				"CAP_DAC_OVERRIDE",
+				"CAP_FSETID",
+				"CAP_FOWNER",
+				"CAP_MKNOD",
+				"CAP_NET_RAW",
+				"CAP_SETGID",
+				"CAP_SETUID",
+				"CAP_SETFCAP",
+				"CAP_SETPCAP",
+				"CAP_NET_BIND_SERVICE",
+				"CAP_SYS_CHROOT",
+				"CAP_KILL",
+				"CAP_AUDIT_WRITE",
+			},
+			Rlimits: []rspec.Rlimit{
+				{
+					Type: "RLIMIT_NOFILE",
+					Hard: uint64(1024),
+					Soft: uint64(1024),
+				},
+			},
+		},
+		Hostname: "shell",
```

Comment:
On Fri, Jul 15, 2016 at 04:25:19PM -0700, Mrunal Patel wrote:
> > +		Hostname: "shell",
> 
> This should be something else. Maybe `example`.

I usually echo the spec's example [1].

[1]: https://github.com/opencontainers/runtime-spec/blame/v1.0.0-rc1/config.md#L188